crates/sui-core/src/unit_tests/move_integration_tests.rs

```diff
@@ -1832,6 +1833,30 @@ fn test_entry_point_string_vec_error() {
     })
 }
 
+#[test]
+#[cfg_attr(msim, ignore)]
+fn test_object_no_id_error() {
+    run_tokio_test_with_big_stack(async move {
+        let mut build_config = BuildConfig::default();
+        build_config.config.test_mode = true;
+        let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        // in this package object struct (NotObject) is defined incorrectly and publishing should
+        // fail (it's defined in test-only code hence cannot be checked by transactional testing
+        // framework which goes through "normal" publishing path which excludes tests).
+        path.push("src/unit_tests/data/object_no_id/");
+        let res = sui_framework::build_move_package(&path, build_config);
+        match res.err().unwrap() {
+            SuiError::ExecutionError(err_str) => {
+                assert!(
+                    err_str.contains("SuiMoveVerificationError")
```

Comment:
might be best to make this two separate asserts

Reply:
Went with the cool pattern on this one!